everestrs/everestrs/src/lib.rs

```diff
@@ -105,13 +110,12 @@ struct Args {
 /// details of the current module, i.e. it deals with JSON blobs and strings as command names. Code
 /// generation is used to build the concrete, strongly typed abstractions that are then used by
 /// final implementors.
-// NOCOM(#sirver): This should be Sync, but I did not find how. It is used by everest from multiple
-// threads, so it must be sync.
-pub trait GenericModule {
+pub trait Subscriber: Sync {
     /// Handler for the command `name` on `implementation_id` with the given `parameters`. The return value
     /// will be returned as the result of the call.
     fn handle_command(
         &self,
+        rt: &Runtime,
```

Comment:
this is the major idea here - I've checked how the actor frameworks are coping with this problem and actix and ractor are passing a ref to which allows calling the publishers into the callback.